### ISSUE TITLE
Exclusions with portals

### ISSUE THREAD

**Issue by isaacminogue** [danielmoi/react-clickout#2]
Hey, just wondering, is it possible to use `data-react-clickout="exclude"` with portals?

Basically, I'm wanting the click outside function to close the dropdown, except when clicking another dropdown (portal). Each portal is wrapped with `wrapWithClickout()`

Currently I'm attaching that attribute to every portal that's being rendered, but that doesn't seem to work 🤔  Does it need to be a standard child or something? Eg, it might not be in the DOM yet when you're making the exclusions.

**Comment by danielmoi** [comment]
Hey, I'm not sure about the solution – and I haven't used this library with portals before.
Exclusion from the clickout depends on the target having that `data-react-clickout` attribute set to `exclude`, I don't think it would matter if that element was created by a portal or not.
So I think it should work.
Do you have a codepen / code sample?

**Comment by isaacminogue** [comment]
Hey again! I ended up using [react-onclickoutside](https://github.com/Pomax/react-onclickoutside) as I couldn't get it to work with yours. I think it's because portals only get added after click, so they're not there in the DOM to be indexed? Not sure. Anyway, I added exclusions in their `handleClickOutside` function where I check if the clicked node is any of the things I want to exclude

```
if (excludeSelf || excludeDropdown || excludeModal) {
 return
}
closeDropdown()
```

I'll have another go with yours sometime and post back if I'm successful, feel free to close the issue though.
